BDArmory/Control/MissileFire.cs

```diff
@@ -6960,20 +7092,21 @@ public void SendTargetDataToMissile(MissileBase ml, bool clearHeat = true)
                             {
                                 if (vesselRadarData.locked) //grab target from pirmary lock
                                 {
-                                    validTarget = true;
                                     targetVessel = vesselRadarData.lockedTargetData.targetData.vessel;
+                                    validTarget = true;
                                 }
                                 else if (irsts.Count > 0) //or brightest ping on IRST
                                 {
-                                    validTarget = true;
                                     targetVessel = vesselRadarData.activeIRTarget(null, this).vessel;
+                                    validTarget = targetVessel;
                                 }
+                                vesselRadarData.LastMissile = ml;
```

Comment:
Shouldn't this be inside 7093's if block? No need for this to be set if no radar lock.